### PR TITLE
Add backwards-compatibility setter and test for `timestamp` attribute

### PR DESCRIPTION
**Pull request by zigaLuksic**

### CODE REVIEW

File: core/eolearn/tests/test_eodata.py [hunk 1]
```diff
@@ -385,3 +386,16 @@ def test_timestamp_consolidation() -> None:
     assert np.array_equal(mask[1:-1, ...], eop.mask["MASK"])
     assert np.array_equal(scalar[1:-1, ...], eop.scalar["SCALAR"])
     assert np.array_equal(mask_timeless, eop.mask_timeless["MASK_TIMELESS"])
+
+
+def test_timestamps_deprecation():
```
[the remark this answers is not in the document]
Comment (zigaLuksic) [reply core/eolearn/tests/test_eodata.py:391]: :thinking: 
knowing the implementation it's redundant. But in tests we shouldn't rely on implementation details, good call

**Review by mlubej — APPROVED** [review]

**Comment by mlubej** [comment]
Some relevant tests still fail in `eo-grow` due to the fact that eopatch stats in json format are compared instead of eopatches. With eopatches the `timestamp` attribute would be converted to `timestamps`, but with json comparison we would need to change the tests files on eo-grow, so not sure if it could/should be handled here.

Other failing tests are due to the non-official changes related to the `FilesystemDataInfo` object, which, upon discussing internally, can be ignored for now.

**Comment by zigaLuksic** [comment]
> Some relevant tests still fail in `eo-grow` due to the fact that eopatch stats in json format are compared instead of eopatches. With eopatches the `timestamp` attribute would be converted to `timestamps`, but with json comparison we would need to change the tests files in eo-grow, so not sure if it could/should be handled here.
> 
> Other failing tests are due to the non-official changes related to the `FilesystemDataInfo` object, which, upon discussing internally, can be ignored for now.

The first sounds like an `eo-grow` issue, so we shouldn't try and fix it here. The second is the problem due to codebreaking changes for `walk_filesystem`. We can create a backport of it, if you think it would be wise, i.e. that `walk_filesystem` remains similar to what it was, and we rename the new one to something else (and `walk_filesystem` is then just the 'reverter').

**Comment by mlubej** [comment]
> The first sounds like an `eo-grow` issue, so we shouldn't try and fix it here. 

agreed

> The second is the problem due to codebreaking changes for `walk_filesystem`. We can create a backport of it, if you think it would be wise, i.e. that `walk_filesystem` remains similar to what it was, and we rename the new one to something else (and `walk_filesystem` is then just the 'reverter').

It all depends on how long we're willing to live with breaking tests on `develop`. Breaking tests might be annoying if there is going to be development on `eo-grow`. On the other hand, since this is currently being worked on, we might as well be patient a bit and fix everything on `eo-grow` then. You call the shots.

**Comment by zigaLuksic** [comment]
> It all depends on how long we're willing to live with breaking tests on develop. Breaking tests might be annoying if there is going to be development on eo-grow. On the other hand, since this is currently being worked on, we might as well be patient a bit and fix everything on eo-grow then. You call the shots.

Not sure what you mean? Tests in `eo-grow` pass just fine, since they are using the latest released version.

**Comment by mlubej** [comment]
> Not sure what you mean? Tests in eo-grow pass just fine, since they are using the latest released version.

Discussed in private, issue was I was using the `develop` version of eo-learn in combination with testing `eo-grow`, instead of the latest release, like it's done in eogrow CI.


> We can create a backport of it, if you think it would be wise, i.e. that walk_filesystem remains similar to what it was, and we rename the new one to something else (and walk_filesystem is then just the 'reverter').

After discussion, will tackle this in a separate MR.

**Review by mlubej — APPROVED** [review]

**Pull request merged**